packages/bff/src/services/agreementService.ts

```diff
@@ -93,6 +97,145 @@ export function agreementServiceBuilder(clients: PagoPAInteropBeClients) {
 
       return enhanceAgreement(agreement, clients, ctx);
     },
+
+    async getAgreementsEserviceProducers({
+      ctx,
+      offset,
+      limit,
+      requesterId,
+      states,
+      eServiceName,
+    }: {
+      offset: number;
+      limit: number;
+      requesterId: string;
+      states: agreementApi.AgreementState[];
+      ctx: WithLogger<BffAppContext>;
+      eServiceName?: string;
+    }): Promise<bffApi.CompactEServicesLight> {
```

Comment:
Could we isolate the context from the rest of the arguments?
So in this case should be something like
```suggestion
    async getAgreementsEserviceProducers({
      offset,
      limit,
      requesterId,
      states,
      eServiceName,
    }: {
      offset: number;
      limit: number;
      requesterId: string;
      states: agreementApi.AgreementState[];
      eServiceName?: string;
    }, { headers, logger }: WithLogger<BffAppContext>): Promise<bffApi.CompactEServicesLight> {
```